pygaggle/run/robust04_reranker_pipeline_gpu.py

```diff
@@ -0,0 +1,193 @@
+                            ' from the first stage retrieval (e.g., BM25).')
+    parser.add_argument('--corpus', type=str, required=True,
+                        help='path to a single file containing all TREC Disks 4 and 5 documents.')
+    parser.add_argument('--output_monot5', type=str, required=True,
+                        help='Path to store the TREC-formatted monoT5 reranked output.')
+    parser.add_argument('--max_length', type=int, default=10,
+                        help='Maximum number of sentences of each segment.')
+    parser.add_argument('--stride', type=int, default=5,
+                        help='Stride (step) in sentences between each segment.')
+    return parser.parse_args(args)
+
+args = parse_args(sys.argv[1:])
+
+# Model
+reranker =  MonoT5()
+monot5_results = args.output_monot5
+
+# Sentencizer
+nlp = spacy.blank("en")
+nlp.add_pipe('sentencizer')
+
+# Input Files
+queries = load_queries(path=args.queries)
+run = load_run(path=args.run)
+corpus = load_corpus(path=args.corpus)
+
+# Pipeline
+n_segments = 0
+n_docs = 0
+n_doc_ids_not_found = 0
+for query_id, doc_ids in tqdm(run.items(), total=len(run)):
+    print(f'{query_id}: Converting to segments...')
+    query_text = queries[query_id]
+    passages = []
+    for doc_id in doc_ids:
+        if doc_id not in corpus:
+            n_doc_ids_not_found += 1
+            continue
+        n_docs += 1
+        doc_text = corpus[doc_id]
+        doc = nlp(doc_text[:10000])
+        sentences = [str(sent).strip() for sent in doc.sents]
+        for i in range(0, len(sentences), args.stride):
+            segment = ' '.join(sentences[i:i + args.max_length])
+            passages.append([doc_id, segment])
+            n_segments += 1
+            if i + args.max_length >= len(sentences):
+                break
+
+    print(f'{query_id}: Reranking...')
+
+    # Reranker using pygaggle
+    query = Query(query_text)
+    texts = [ Text(p[1], {'docid': p[0]}, 0) for p in passages]
+    start = time()
+    ranked_results = reranker.rerank(query, texts)
+    end = time()
+    elapsed_time = end - start
+    print("Time Elapsed: {:.1f}".format(elapsed_time))
+
+    # Get scores from reranker
+    final_t5_scores = {}
+    for result in ranked_results:
+        if result.metadata["docid"] not in final_t5_scores:
+            final_t5_scores[result.metadata["docid"]] = result.score
+        else:
+            if final_t5_scores[result.metadata["docid"]] < result.score:
+                final_t5_scores[result.metadata["docid"]] = result.score
+
+    # Writes a run file in the TREC format
+    for rank, (docid, score) in enumerate(final_t5_scores.items()):
+        with open(monot5_results, mode='a') as writer:
+            writer.write(f'{query_id} Q0 {docid} {rank + 1} {1 / (rank + 1)} T5\n')
+
+print(f'Wrote {n_segments} segments from {n_docs} docs.')
+print(f'{n_doc_ids_not_found} doc ids not found in the corpus.')
+print('Done!')
```

Comment:
Add an empty line at the end of the file

Reply:
Is a "/n" line ok?